BacktraceScreen.c

```diff
@@ -0,0 +1,211 @@
+/*
+htop - Action.c
+(C) 2015 Hisham H. Muhammad
+Released under the GNU GPLv2+, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include "BacktraceScreen.h"
+
+#if (defined(HTOP_LINUX) && defined(HAVE_LIBUNWIND_PTRACE))
+
+#include <sys/wait.h>
+
+#include "CRT.h"
+#include "Object.h"
+#include "Panel.h"
+#include "Process.h"
+#include "RichString.h"
+#include "XUtils.h"
+#include "errno.h"
+
+#ifdef HAVE_LIBIBERTY
+#include <libiberty/demangle.h>
+#endif
+
+#include <libunwind-ptrace.h>
+#include <sys/ptrace.h>
+
+#define MAX_FRAME 256
+
+static const char* const BacktraceScreenFunctions[] = {"Done   ", NULL};
+
+static const char* const BacktraceScreenKeys[] = {"Esc"};
+
+static const int BacktraceScreenEvents[] = {27};
+
+static void Frame_display(const Object* super, RichString* out) {
+   const Frame* const frame = (const Frame*)super;
+   if (frame->isError) {
+      RichString_appendAscii(out, CRT_colors[DEFAULT_COLOR], frame->error);
+      return;
+   }
+
+   char bufferNumberOfFrame[16] = {'\0'};
+   int len = snprintf(bufferNumberOfFrame, sizeof(bufferNumberOfFrame), "#%-3d ", frame->index);
+   RichString_appendnAscii(out, CRT_colors[DYNAMIC_GREEN], bufferNumberOfFrame, len);
+
+   char bufferAddress[32] = {'\0'};
+   len = snprintf(bufferAddress, sizeof(bufferAddress), "0x%016zx ", frame->address);
+   RichString_appendnAscii(out, CRT_colors[DYNAMIC_BLUE], bufferAddress, len);
+
+   RichString_appendAscii(out, CRT_colors[DEFAULT_COLOR], frame->functionName);
+   if (frame->isSignalFrame) {
+      RichString_appendAscii(out, CRT_colors[DYNAMIC_RED], " signal frame");
+   }
+
+   char bufferFrameOffset[16] = {'\0'};
+   len = snprintf(bufferFrameOffset, sizeof(bufferFrameOffset), "+%zu", frame->offset);
+   RichString_appendAscii(out, CRT_colors[DYNAMIC_YELLOW], bufferFrameOffset);
+}
+
+static void BacktracePanel_getFrames(BacktracePanel* this) {
+   Panel* super = (Panel*) this;
+
+   unw_addr_space_t addrSpace = unw_create_addr_space(&_UPT_accessors, 0);
+   if (!addrSpace) {
+      xAsprintf(&this->error, "Unable to init libunwind.");
+      return;
+   }
+
+   const pid_t pid = Process_getPid(this->process);
+
+   if (pid == 0) {
```

Comment:
Is it intentional to compare `(pid == 0)` only or is it `(pid <= 0)`? (PID is a signed type. Negative PIDs are reserved for references to process groups in most Unix-like systems.)

Reply:
Okay, If I understand correctly what you said, a negative PID can represent a group of processes, but can `Process_getPid` return a group of processes (and return a negative pid)? 
But It makes sense to attach to one process/thread and not a group of processes, so it should be `pid <= 0`.